jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -226,44 +210,50 @@ public static boolean isValidJavaClass(String className) {
    * @throws NumberFormatException if any of the ports are in an invalid format or out of range
    */
   @VisibleForTesting
-  static ImmutableSortedMap<String, EmptyStruct> portListToPortMap(List<String> ports)
-      throws NumberFormatException {
-    SortedMap<String, EmptyStruct> result = new TreeMap<>();
+  static ImmutableList<String> expandPortRanges(List<String> ports, BuildLogger logger) {
+    List<String> result = new ArrayList<>();
 
     for (String port : ports) {
       // Remove all spaces
-      port = port.trim().replace(" ", "");
+      port = port.replace(" ", "");
 
       if (port.matches("\\d+")) {
         // Port is a single number
         int portNum = Integer.parseInt(port);
         if (portNum < 1 || portNum > 65535) {
-          throw new NumberFormatException("Invalid port number " + port);
+          logger.warn("Port number '" + port + "' is out of range (1-65535)");
         }
-        result.put(port + "/tcp", EmptyStruct.get());
+        result.add(port);
 
       } else if (port.matches("\\d+-\\d+")) {
         // Port is a range
         List<String> range = Splitter.on('-').splitToList(port);
         int min = Integer.parseInt(range.get(0));
         int max = Integer.parseInt(range.get(1));
 
+        // Reverse range if configured as 'max-min' instead of 'min-max'
+        if (min > max) {
```

Comment:
I feel like we should prob error in this case instead of fixing it secretly for them, since it might be more likely they typed the wrong port numbers than wrote them going from high to low.